docs/book/src/developer/providers/v1alpha4-to-v1beta1.md

```diff
@@ -47,6 +47,9 @@ v1alpha4, the same conversions have to be implemented.
 - **ParseMajorMinorPatch** has been removed in favor of `ParseMajorMinorPatch` in the util/version package.
 - **GetMachinesForCluster** has been removed in favor of `GetFilteredMachinesForCluster` in the util/collection package.
 - **GetControlPlaneMachines** has been removed in favor of `FromMachines(machine).Filter(collections.ControlPlaneMachines(cluster.Name))`  in the util/collection package.
-- **GetControlPlaneMachinesFromList** has been removed in favor of `FromMachineList(machines).Filter(collections.ControlPlaneMachines(cluster.Name))` in the util/collection package. 
+- **GetControlPlaneMachinesFromList** has been removed in favor of `FromMachineList(machines).Filter(collections.ControlPlaneMachines(cluster.Name))` in the util/collection package.
 - **GetCRDMetadataFromGVK** has been removed in favor of `GetGVKMetadata`.
+- Ensure your template resources support `template.meta` fields. Refer to the [cluster][cluster-contract] and [machine][machine-contract] provider contract docs for more information.
```

Comment:
Nit:
Should we add some more color why this is required.
As far as I understand this problem space:
- This requirement is about adding some infrastructure specific metadata to infrastructure templates (and CAPI will just preserve them when creating resources derived from templates).
- Metadata propagation IMO is a somehow related feature that makes metadata flow down from high level abstraction KCP/MD to generated objects. This requirement does not impact how this feature works.

Reply:
I think @MaxRink and @yastij have the specifics of why you would want it.

> This requirement does not impact how this feature works. 
Yes, that's correct.

Reply:
Clarification added.